h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/mapred/H2OYarnClientProtocolProvider.java

```diff
@@ -0,0 +1,36 @@
+package water.hadoop.mapred;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.MRConfig;
+import org.apache.hadoop.mapreduce.protocol.ClientProtocol;
+import org.apache.hadoop.mapreduce.protocol.ClientProtocolProvider;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+
+/**
+ * H2O specific yarn client provider.
+ *
+ * The provider can be selected by providing: `-Dmapreduce.framework.name=h2o-yarn`
+ */
+public class H2OYarnClientProtocolProvider extends ClientProtocolProvider {
+
+  @Override
+  public ClientProtocol create(Configuration conf) throws IOException {
+    if ("h2o-yarn".equals(conf.get(MRConfig.FRAMEWORK_NAME))) {
```

Comment:
What are the naming conventions here? I know that Apache Tez uses "yarn-h2o". Not sure if there are other examples like that. Perhaps we should use "-h2o" as a suffix instead.